mlir/lib/Quantum/IR/QuantumOps.cpp

```diff
@@ -40,6 +40,35 @@ static const mlir::StringSet<> hermitianOps = {"Hadamard", "PauliX", "PauliY", "
                                                "CY",       "CZ",     "SWAP",   "Toffoli"};
 static const mlir::StringSet<> rotationsOps = {"RX",  "RY",  "RZ",  "PhaseShift",
                                                "CRX", "CRY", "CRZ", "ControlledPhaseShift"};
+
+LogicalResult StaticCustomOp::canonicalize(StaticCustomOp op, mlir::PatternRewriter &rewriter)
+{
+    if (op.getAdjoint()) {
```

Comment:
Yes I agree with this. I think I just copied over customOp canonicalize method here. Didn't look too much into it :) I will fix this as well

Reply:
> > This looks a lot better! You'll also need to change the merge rotations and cancel inverses passes right?
> Thanks @erick-xanadu, yes I am planning to change those today.

I also don't think these need to be updated to run on the static ops since they can run after the conversion to dynamic.

Reply:
With the current design, the lowering is done after the apply-transform-sequence pass which is ultimately the place where peephole optimizations are run, so passes like cancel inverses and merge rotations should be able to deal with static custom ops. I think the reason for this placement in pipeline is to allow external contributors to schedule their passes within the apply-transform-sequence pass.

Reply:
Ah I see, oh well then I guess it's fine 👍